### PR TITLE
Update README adding Contribution guidelines and tweak Deploy Nextjs doc

### PR DESCRIPTION
**Pull request by Pabl0cks**
Solves #5 and #11

### CODE REVIEW

**Comment by Pabl0cks** [comment]
Can you take a look when you have a moment @technophile-04? 🙌

**Comment by Pabl0cks** [comment]
I've just pushed a change to tweak where users land after they "Edit this page" button, so they open the edition form instead. 

I've seen more projects sending to that url than sending just to the read version of the file (/blob/)

File: README.md [hunk 1]
```diff
@@ -1,41 +1,82 @@
...
+## Installation
+
+Clone the repository, install all dependencies as well as build all local packages, and then start Docusaurus.
 
 ```
-$ yarn start
+git clone https://github.com/scaffold-eth/se2-docs.git
+cd se2-docs
+yarn install
+yarn start
```
Comment (technophile-04) [inline README.md:17]: We are using `npm` as our package manager for this, we should migrate to `yarn` in future

Lets update this for now to : 

```shell
npm install

npm run start
```

File: README.md [hunk 2]
```diff
@@ -1,41 +1,82 @@
...
+-->
+
+<!---
+TODO: Explain _category_.json
+-->
+
+## Website contributions
+
+For contributions to the Website Code, fork the repository, create a new branch, and make all changes necessary in your repository. Once you are done with your changes, create a PR to Scaffold-ETH 2 Documentation repository.
+
```
Comment (technophile-04) [inline README.md:65]: Should we mention about this : ["fork-and-pull" Git workflow](https://github.com/susam/gitpr) 

Like you can learn more about it here : ["fork-and-pull" Git workflow](https://github.com/susam/gitpr) ? 

we had this in SE-2 contribution.md -> https://github.com/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md#pull-requests|

But feel free to omit it 🙌

File: docs/deploying/deploy-nextjs-app.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# Deploy your NextJS App
+
+:::tip Hint
+We recommend connecting your GitHub repo to Vercel (through the Vercel UI) so it gets automatically deployed when pushing to `main`.
+:::tip Hint
+
+If you want to deploy directly from the CLI, run this and follow the steps to deploy to Vercel:
+
+```
+`yarn vercel`
```
Comment (technophile-04) [inline docs/deploying/deploy-nextjs-app.mdx:14]:
```suggestion
yarn vercel
```

File: docs/deploying/deploy-nextjs-app.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+Once you log in (email, github, etc), the default options should work. It'll give you a public URL.
+
+If you want to redeploy to the same production URL you can run:
+
+```
+`yarn vercel --prod`
```
Comment (technophile-04) [inline docs/deploying/deploy-nextjs-app.mdx:22]:
```suggestion
yarn vercel --prod
```

Same here

**Comment by technophile-04** [comment]
Great work Pablo the Readme looks great to me !!! 

---

Regarding #11  : 
>  if you see that configuration attributes are too big now in the content, we can always roll back those changes.
> image

Yeah I agree they are too big :(, 

What if we add this [`toc_max_heading_level`](https://docusaurus.io/docs/next/markdown-features/toc#table-of-contents-heading-level) in `deploy-nextjs-app.mdx` ? 
```mdx
---
sidebar_position: 2
toc_max_heading_level: 5
---
```

And then make attributes `h4`? It looks something like this : 

Screenshot 2023-08-11 at 1 00 32 AM

File: README.md (hunk 1 above)
Comment (Pabl0cks) [reply README.md:17]: Ups! Saw it in the old Readme and always used it, so never questioned hah 

Now I've realized there is no yarn.lock file in the repo, updating! 👍

File: docs/deploying/deploy-nextjs-app.mdx (hunk 3 above)
Comment (Pabl0cks) [reply docs/deploying/deploy-nextjs-app.mdx:14]: No idea why I added them, fixing! Thanks 🧐

**Comment by Pabl0cks** [comment]
> What if we add this [toc_max_heading_level](https://docusaurus.io/docs/next/markdown-features/toc#table-of-contents-heading-level) in deploy-nextjs-app.mdx ?

Wow I saw them not showing and didn't think about them having an option to show that depth in the right sidebar. I thought would be a problem of h3 not existing. Good job!!

I have pushed all the changes, nice improvements/bug catches! Thanks for reviewing so quick 🙌

**Comment by technophile-04** [comment]
Tysm Pablo everything looking nice !!! Merging this 🙌

**Pull request merged**